cartoframes/contrib/vector.py

```diff
@@ -245,16 +278,30 @@ def _get_html_doc(sources,
         airship_styles_path = _airship_path + _AIRSHIP_STYLE
         airship_icons_path = _airship_path + _AIRSHIP_ICONS_STYLE
 
-    return srcdoc.replace('@@SOURCES@@', json.dumps(sources)) \
-        .replace('@@BASEMAPSTYLE@@', basemap) \
-        .replace('@@MAPBOXTOKEN@@', token) \
-        .replace('@@CREDENTIALS@@', json.dumps(credentials)) \
-        .replace('@@BOUNDS@@', bounds) \
-        .replace('@@CARTO_VL_PATH@@', _carto_vl_path) \
-        .replace('@@AIRSHIP_COMPONENTS_PATH@@', airship_components_path) \
-        .replace('@@AIRSHIP_BRIDGE_PATH@@', airship_bridge_path) \
-        .replace('@@AIRSHIP_STYLES_PATH@@', airship_styles_path) \
-        .replace('@@AIRSHIP_ICONS_PATH@@', airship_icons_path)
+    camera = None
+    if viewport is not None:
+        camera = {
+            'center': _get_center(viewport),
+            'zoom': viewport.get('zoom'),
+            'bearing': viewport.get('bearing'),
+            'pitch': viewport.get('pitch')
+        }
+
+    return template.render(
+        width=width,
+        height=height,
+        sources=json.dumps(sources),
```

Comment:
Is it better to pass `sources` and `credentials` as a `dict` and add the `tojson` filter in the jinja template like you've done for `camera`?